R/FilteredData-utils.R

```diff
@@ -32,50 +32,32 @@ init_filtered_data.TealData <- function(x, # nolint
                                         code = x$get_code_class(),
                                         cdisc = FALSE,
                                         check = x$get_check()) {
+  cdisc <- if (length(join_keys$get_parents()) > 0) TRUE else FALSE
   data_objects <- lapply(x$get_datanames(), function(dataname) {
     dataset <- x$get_dataset(dataname)
-    list(
-      dataset = dataset$get_raw_data(),
-      keys = dataset$get_keys(),
-      metadata = dataset$get_metadata(),
-      label = dataset$get_dataset_label()
-    )
-  })
-  names(data_objects) <- x$get_datanames()
-
-  init_filtered_data(
-    x = data_objects,
-    join_keys = join_keys,
-    code = code,
-    check = check
-  )
-}
 
-#' @keywords internal
-#' @export
-init_filtered_data.CDISCTealData <- function(x, # nolint
-                                             join_keys = x$get_join_keys(),
-                                             code = x$get_code_class(),
-                                             cdisc = TRUE,
-                                             check = x$get_check()) {
-  data_objects <- lapply(x$get_datanames(), function(dataname) {
-    dataset <- x$get_dataset(dataname)
-
-    # CDISCTealData can contain TealDataset and CDISCTealDataset objects
-    # the former do not have a get_parent() call
     parent <- if (inherits(dataset, "CDISCTealDataset")) {
       dataset$get_parent()
     } else {
       NULL
     }
 
-    list(
-      dataset = dataset$get_raw_data(),
-      keys = dataset$get_keys(),
-      metadata = dataset$get_metadata(),
-      label = dataset$get_dataset_label(),
-      parent = parent
-    )
+    if (cdisc) {
+      list(
+        dataset = dataset$get_raw_data(),
+        keys = dataset$get_keys(),
```

Comment:
I think it's fine to get keys like this for now, in the future we'll get them directly from the join_keys - similarly for parent in line 40 we'll get from join_keys when creating tdata object.

Reply:
yes that step can be done with the tdata switch.